### PR TITLE
PR for rails form Project

### PR DESCRIPTION
**Pull request by TSHEPO-CLOUD**
The features of this project are listed as below:

- Create User Model
- Create Routes
- Create Forms
- Create controllers
- Create Views

### CODE REVIEW

**Review by oloomoses — CHANGES_REQUESTED** [review]
## Hello team, you are on the right track with this project so far. :clap: 
## Work out on the suggested changes before submitting another review.

- [ ] Ensure it is possible to edit and update a user in the app [Link](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/edit_user.png)
- [ ] Update your routes and controller to handle editing an existing user
- [ ] Ensure the `create` action users `user_params` to initialize user [link](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png)
- [ ] Ensure you have the `edit`  action as per the project requirements. [Link](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png)
- [ ] Add the `user_params` method with the correct fields

### Views
- [ ] Ensure you have created all the three forms with `form` and `form_tag` commented out
- [ ] Ensure you implement the `edit` form as per the requirements [Link](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_edit.png)

## Optional
- [ ] Create partial for user form and reuse it edit and new view
- [ ] form view to displays a list of the error messages
- [ ] validate the email format by using either a regular expression or a built-in URI method.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (rindrajosia) [inline app/controllers/users_controller.rb]: Remove comments

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  <label for="username">Username</label>
+  <input type="text" name="user[username]" id="username">
+  <label for="email">Email</label>
+  <input type="email" name="user[email]" id="email">
+  <label for="password">Password</label>
+  <input type="password" name="user[password]" id="password">
+  <input type="submit" value="Submit">
+</form> -->
+
+<!--<%= form_tag("/users", method: "post") do %>
```
Comment (rindrajosia) [inline app/views/users/new.html.erb]: It's preferable to use **users_path**

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<h1>Welcome...</h1>
+
+<!--<form action="/users" method="post" accept-charset="UTF-8">
```
Comment (rindrajosia) [inline app/views/users/new.html.erb]: It's preferable to use **users_path**

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+<% end %>-->
+
+<%= form_for @user do |f| %>
+  <%= f.label :username %>
+  <%= f.text_field :username, placeholder: "user" %>
+  <%= f.label :email %>
+  <%= f.email_field :email, placeholder: "example@example.com"%>
+  <%= f.label :password %>
+  <%= f.password_field :password %>
+  <%= f.submit%>
```
Comment (rindrajosia) [inline app/views/users/new.html.erb]: [OPTIONAL] Change this line to **<%= f.submit :Submit%>**

File: app/views/users/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<h1>Edit user...</h1>
+
+<%= form_for(@user) do |f| %>
+  <%= f.label :username %>
+  <%= f.text_field :username, placeholder: "user" %>
+  <%= f.label :email %>
+  <%= f.email_field :email, placeholder: "john@doe.com "%>
+  <%= f.label :password %>
+  <%= f.password_field :password %>
+  <%= f.submit%>
```
Comment (rindrajosia) [inline app/views/users/edit.html.erb]: [OPTIONAL] Change this line to: **<%= f.submit :Submit%>**

**Review by rindrajosia — CHANGES_REQUESTED** [review]
**Changes required**
Hello team!!!

You did a very good job.
However, to move forward you have to make some changes.

Kindly read the comments below, then make the suggested changes and submit a new code review request
Thanks

Rindra Josia

**Review by peterrobert — APPROVED** [review]
### APPROVED 👍🏿 💯 🥇 
Hello, @TSHEPO-CLOUD very nice work. Good job implementing all the required changes. Also, your application meets all the requirements. 👏🏿 

### [OPTIONAL]

- [ ] It's actually good practice to create partials. The new view part of the application and the edit view part of the application uses the same form structure. It is at this point that you should utilize the power of rails partials. Read more on this [here](https://guides.rubyonrails.org/layouts_and_rendering.html)

**If you have a question reach me on slack @peterrobert**
### on to the next project and HAPPY CODING.

**Pull request merged**